### ISSUE TITLE
Removing BatchNormalization

### ISSUE THREAD

**Issue by v-r-x** [szagoruyko/cifar.torch#4]
Hi,

I am using the nn version but removed BatchNormalization and ceil() and I am getting this error:

/home/vrx/torch/install/bin/luajit: ...vrx/torch/install/share/lua/5.1/nn/SpatialMaxPooling.lua:18: aborting at /home/vrx/torch/extra/cunn/SpatialMaxPooling.cu:261
stack traceback:
    [C]: in function 'SpatialMaxPooling_updateOutput'
    ...wig/torch/install/share/lua/5.1/nn/SpatialMaxPooling.lua:18: in function 'updateOutput'
    /home/vrx/torch/install/share/lua/5.1/nn/Sequential.lua:39: in function 'updateOutput'
    /home/vrx/torch/install/share/lua/5.1/nn/Sequential.lua:39: in function 'updateOutput'
    /home/vrx/torch/install/share/lua/5.1/nn/Sequential.lua:39: in function 'forward'
    train_rgb.lua:107: in function 'opfunc'
    /home/vrx/torch/install/share/lua/5.1/optim/sgd.lua:43: in function 'sgd'
    train_rgb.lua:116: in function 'train'
    train_rgb.lua:167: in main chunk
    [C]: in function 'dofile'
    ...vrx/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
    [C]: at 0x00406670
error in SpatialMaxPooling.updateOutput: invalid argument

If I change the batch size from 128 to 64 it works, but the training accuracy does not improve (values between 9.77% and 10.14%) and the test accuracy stays exactly at 10%.

With BatchNormalization I get 67% test accuracy after 9 epochs (using rgb images preprocessed only by subtracting per pixel means and dividing stds).
